tests/languages/docker_test.py

```diff
@@ -1,14 +1,134 @@
+import builtins
+import json
+from typing import List
 from unittest import mock
+from unittest.mock import mock_open
+
+import pytest
 
 from pre_commit.languages import docker
 
 
 def test_docker_fallback_user():
     def invalid_attribute():
         raise AttributeError
+
     with mock.patch.multiple(
-        'os', create=True,
-        getuid=invalid_attribute,
-        getgid=invalid_attribute,
+            'os', create=True,
+            getuid=invalid_attribute,
+            getgid=invalid_attribute,
     ):
         assert docker.get_docker_user() == ()
+
+
+class TestInDocker:
+
+    @pytest.fixture
+    def mock_file_fixture(self):
+        return lambda read_data: \
```

Comment:
no backslashes, again

you probably want this to  be a yield fixture:

```
@pytest.fixture
def mocked_read():
    with mock.patch.object(builtins, 'open', new_callable=mock_open) as mck:
        yield mck
```

and then in your test, utilize `mocked_read.read_data = ...`

Reply:
>you probably want this to be a yield fixture:

Not really, `yield` one doesn't really work, it doesn't allow to set `read_data` afterwards. Seems like it can be set only in constructor, so I keep it like it is.